sdk/storage/azure_storage_blob/tests/page_blob_client.rs

```diff
@@ -28,10 +29,7 @@ async fn test_create_page_blob(ctx: TestContext) -> Result<(), Box<dyn Error>> {
     assert_eq!(BlobType::PageBlob, blob_type.unwrap());
 
     // Create If Not Exists Scenario
-    let create_options = PageBlobClientCreateOptions {
-        if_none_match: Some("*".to_string()),
-        ..Default::default()
-    };
+    let create_options = page_blob_create_if_not_exists(PageBlobClientCreateOptions::default());
```

Comment:
Wondering if there is a better way to achieve this effect? We would like to abstract out the logic to get this to be "create if not exists". 

Even better if we could make this generic, as this is likely not the only API that would support "if not exists". Maybe a Trait of some sort and we somehow mark all the applicable option bags with this Trait?

Reply:
Closed as duplicate on megathread on helpers.